front/src/features/tables/components/ColumnHeader.tsx

```diff
@@ -0,0 +1,49 @@
+import { useNavigation } from "@/hooks/useNavigation";
+import { SortOptions } from "@/types/logDisplayOptions";
+import { SortIcon } from "./SortIcon";
+
+type ColumnHeaderProps = {
+    title: string;
+    sortKey?: SortOptions;
+    width?: "small" | "medium" | "large";
+};
+
+export const ColumnHeader = ({ title, sortKey, width }: ColumnHeaderProps) => {
+    const { sortBy, sortOrder, updateSearchParam } = useNavigation();
+    let order: string | undefined = undefined;
+    if (sortBy === sortKey) {
+        order = sortOrder;
+    }
+
+    const updateSort = (key: SortOptions) => {
+        const newOrder =
+            order === undefined || order === "asc" ? "desc" : "asc";
+        updateSearchParam({ sortBy: key, sortOrder: newOrder });
+    };
+
+    let thClass = "px-4 py-2 text-left text-gray-600 border-gray-300 border-x";
```

Comment:
not sure whether this will work, have you checked?

see [here](https://v2.tailwindcss.com/docs/just-in-time-mode#arbitrary-value-support) under "Don't use string concatenation to create class names"